pkg/ip/Dockerfile

```diff
@@ -8,6 +8,9 @@ RUN apk add --no-cache --initdb -p /out \
     musl \
     wireguard-tools
 
+# We grab our version of wg-quick, called lk-wg-config.sh
+RUN wget -O /usr/bin/lk-wg-config https://gist.githubusercontent.com/zx2c4/3624de869ab7eaef3de5ea8f2b867be9/raw/de72b018f4f4548858ce6aae2898b34db0474221/lk-wg-config.sh
```

Comment:
I don't think moving the file behind a webserver to be fetched at build time changes anything wrt the GPL obligations upon the resulting image which is being built (which AIUI is the objection to the script, although perhaps I was mistaken).

Reply:
In that case, you're not allowed to use any GPL software in any images? Which obviously isn't the case, considering the use of bash, Linux, etc in some resultant images.

Reply:
No, we need it to be easy for people to get the source for the GPL code they are using. The way we have been doing this is to make sure it is all in Alpine packages or kernel source, where we can easily trace back from the source and tags to which source is being shipped.

Reply:
> No, we need it to be easy for people to get the source for the GPL code they are using. 

That's trivially achieved for a shell script like this though, isn't it? It's basically a nop...

Reply:
> No, we need it to be easy for people to get the source

Isn't Github's gist considered to be pretty easy to use?

If not, any suggestions for other easy places to host this file? I can amend this patch set and host elsewhere.

Reply:
Actually, it looks like you're already doing this tons of places using `curl`, so I think this method will be okay. I'll fix up this commit to use `curl` instead of `wget`.

```
test/pkg/ltp/Dockerfile:RUN curl -fsSL -o ltp-full-${LTP_VERSION}.tar.xz ${LTP_SOURCE}
projects/clear-containers/kernel/Dockerfile:RUN curl -fsSL -o linux-${KERNEL_VERSION}.tar.xz ${KERNEL_SOURCE}
projects/selinux/kernel/Dockerfile:RUN curl -fsSL -o linux-${KERNEL_VERSION}.tar.xz ${KERNEL_SOURCE}
projects/landlock/kernel-landlock/Dockerfile:RUN curl -fsSL -o linux-${KERNEL_VERSION}.tar.xz ${KERNEL_SOURCE}
projects/shiftfs/Dockerfile:RUN curl -fsSL -o linux-${KERNEL_VERSION}.tar.xz ${KERNEL_SOURCE}
projects/ima-namespace/kernel/Dockerfile:RUN curl -fsSL -o linux-${KERNEL_VERSION}.tar.xz ${KERNEL_SOURCE}
projects/kernel-config/Dockerfile:RUN curl -fsSL -o linux-${KERNEL_VERSION}.tar.xz ${KERNEL_SOURCE}
projects/memorizer/kernel-memorizer/Dockerfile:RUN curl -fsSLO ${KERNEL_SHA256_SUMS} && \
projects/ebpf/ebpf.build/Dockerfile:RUN curl -sSL -O https://fedorahosted.org/releases/e/l/elfutils/0.165/elfutils-$ELFUTILS_VERSION.tar.bz2
projects/kubernetes/kubernetes/Dockerfile:RUN curl -fSL -o /tmp/cni.tgz https://github.com/containernetworking/cni/releases/download/v0.5.2/cni-amd64-${cni_version}.tgz && \
projects/kubernetes/kubernetes/Dockerfile:RUN curl -fSL -o /out/etc/weave.yaml https://cloud.weave.works/k8s/v1.7/net?v=${weave_version} 
projects/kubernetes/kubernetes/Dockerfile:RUN curl -fSL -o /out/usr/bin/kubelet https://dl.k8s.io/${kubernetes_version}/bin/linux/amd64/kubelet && chmod 0755 /out/usr/bin/kubelet
projects/kubernetes/kubernetes/Dockerfile:RUN curl -fSL -o /out/usr/bin/kubeadm https://dl.k8s.io/${kubernetes_version}/bin/linux/amd64/kubeadm && chmod 0755 /out/usr/bin/kubeadm
projects/kubernetes/kubernetes/Dockerfile:RUN curl -fSL -o /out/usr/bin/kubectl https://dl.k8s.io/${kubernetes_version}/bin/linux/amd64/kubectl && chmod 0755 /out/usr/bin/kubectl
kernel/Dockerfile:RUN curl -fsSLO ${KERNEL_SHA256_SUMS} && \
kernel/Dockerfile:RUN curl -sSL -o /wireguard.tar.xz "${WIREGUARD_URL}" && \
```

Reply:
My point was not that there is any reason to avoid curl/wget per se, just that if the only reason for doing so is to avoid licensing issues the it is not actually making any difference to the situation, the license problems are the same whether the file is directly in this repo or fetched at build time.

So either the licensing situation for distributing this script in an image is fine (from a license compliance PoV) and we can just put this file in this repo and move on or the licensing is not fine and fetching it at runtime does not somehow make it ok.

If there are other places in the code which are also using curl/wget (or any other build time fetch) solely in a (IMHO misguided) attempt to address some licensing issue or other then we should raise issues here so we can confirm whether or not remote fetching makes any difference to the license in question (it seems unlikely to me that it would for any normal OSS license).

Note that there are other legitimate reasons for fetching things at build time.

To be clear, my opinion is that the GPL is pretty trivially satisfiable for a shell script, about the only sticking point would be the last bit of clause 1:
> and give any other recipients of the Program a copy of this License along with the Program.
which we may or may not be doing depending on what is in the base image.

It looks like @justincormack might have a different opinion on that. It also seems he wants to address licensing more generally via https://github.com/moby/tool/issues/147 (and presumably that will also necessitate changes to various LinuxKit package too).

Reply:
> just that if the only reason for doing so is to avoid licensing issues the it is not actually making any difference to the situation, the license problems are the same whether the file is directly in this repo or fetched at build time. So either the licensing situation for distributing this script

I think your premises might be flawed. The objection, in the first place, was that @justincormack didn't want to put GPL files actually inside this repo. I find that a bit silly, but it's not up to me. The solution was therefore just to move the file outside of the repo. Putting GPL code inside the final images has always been okay, and has never been what this discussion is about.